python/lsst/meas/deblender/baseline.py

```diff
@@ -33,50 +34,232 @@
+            self.peakCount = maxNumberOfPeaks
+
+        if filters is None:
+            filters = range(len(footprints))
+        self.filters = filters
+
+        # Create a DeblendedParent for the Footprint in every filter
+        self.deblendedParents = OrderedDict([])
+        for n in range(self.filterCount):
+            f = self.filters[n]
+            dp = DeblendedParent(f, footprints[n], maskedImages[n], psfs[n],
+                                 psffwhms[n], avgNoise[n], maxNumberOfPeaks, self)
+            self.deblendedParents[self.filters[n]] = dp
+
+        # Group the peaks in each color
+        for idx in range(self.peakCount):
+            peakDict = {f: dp.peaks[idx] for f,dp in self.deblendedParents.items()}
+            multiPeak = MultiColorPeak(peakDict, idx, self)
+
+    def getParentProperty(self, propertyName):
+        """Get the footprint in each filter"""
+        return [getattr(fp, propertyName) for dp in self.deblendedParents]
+
+    def setTemplateSums(self, templateSums, fidx=None):
+        if fidx is not None:
+            self.templateSums[fidx] = templateSums
+        else:
+            for f, templateSum in templateSums.items():
+                self.deblendedParents[f].templateSum = templateSum
+
+class DeblendedParent(object):
+    def __init__(self, filterName, footprint, maskedImage, psf, psffwhm, avgNoise, 
+                 maxNumberOfPeaks, debResult):
+        self.filter = filterName
+        self.fp = footprint
+        self.fp.normalize()
+        self.maskedImage = maskedImage
+        self.psf = psf
+        self.psffwhm = psffwhm
+        self.img = maskedImage.getImage()
+        self.varimg = maskedImage.getVariance()
+        self.mask = maskedImage.getMask()
+        self.avgNoise = avgNoise
+        self.updateFootprintBbox()
+        self.debResult = debResult
+        self.peakCount = debResult.peakCount
         self.templateSum = None
+        
+        # avgNoise is an estiamte of the average noise level for the image in this filter
+        if avgNoise is None:
+            stats = afwMath.makeStatistics(self.varimg, self.mask, afwMath.MEDIAN)
+            avgNoise = math.sqrt(stats.getValue(afwMath.MEDIAN))
+            debResult.log.trace('Estimated avgNoise for filter %s = %f', self.filter, avgNoise)
+        self.avgNoise = avgNoise
+        
+        # Store all of the peak information in a single object
+        self.peaks = []
+        peaks = self.fp.getPeaks()
+        for idx in range(self.peakCount):
+            deblendedPeak = DeblendedPeak(peaks[idx], idx, self)
+            self.peaks.append(deblendedPeak)
+
+    def updateFootprintBbox(self):
+        """Update the bounding box of the parent footprint
+        
+        If the parent Footprint is resized, for example when `meas.deblender.baseline.deblend`
+        has `patchEdges=True`, it will be necessary to update the bounding box of the footprint.
```

Comment:
You're right. I moved ``self.imbb = self.img.getBBox()`` to the init function and left only the lines that store the new footprint bounding box.